webpush/encrypt.go

```diff
@@ -139,36 +142,64 @@ type EncryptionResult struct {
 	ServerPublicKey []byte
 }
 
+// Decrypt an encrypted messages.
```

Comment:
This package can be used for sending messages, but it can also be used to implement an UA - the spec doesn't require the UA to be a browser, could be an IoT device.

I'm trying to add the http2 framer and subscribe calls to get to a full-UA status.